예약 페이지 api 연동 및 페이지 수정

heroinesy opened the pull request:
예약 확인페이지, 결제, 결제 db = 세개 다 동일한 가격 ✅

수요자 상세 페이지
- 매니저 리스트 3순위까지 출력 historydto ✅
- reservationStatus : done 일 때 /{id}/comment ✅
  - check-in/out 시간
  - 코멘트
- 매니저 상세 페이지에 시간 추가 ✅
- 예약 취소 ✅
- 취소 후 환불
- 수요자 매칭 거절 ( 환불  or 재매칭 모달 붙이기)

예약 폼
- 추천 시간 api 연동 /recommend-duration
- 카테고리 디자인
- 서비스 시간 제한 해제 -> 12시간으로 늘렸어요 ✅
- 주소 평수 유효성 (숫자만) ✅

@yerim29 
Check-in/out 하면 수요자한테 알림 전송 부탁드릴게용

heroinesy committed:
refact: 결제 페이지에서 결제 금액 다른 이슈 수정

heroinesy committed:
refact: 예약 폼 및 API 타입 수정

- 예약 폼에서 카테고리 가격 및 시간을 표시하는 문구 수정
- 예약 시간 선택 로직에서 최대 시간을 12시간으로 변경
- 예약 관련 API 타입을 reservation.ts로 이동 및 수정
- 불필요한 console.log 제거 및 코드 정리

heroinesy committed:
feat: 주소 입력 및 추가 주소 모달에서 평수 입력 안내 수정

- 주소 입력 페이지에서 평수 라벨에 안내 문구 추가
- 추가 주소 모달에서 평수 placeholder에 숫자 입력 안내 추가
- 불필요한 import 문 제거

heroinesy committed:
refact: remove console.log

heroinesy committed:
feat: 예약 취소 기능 추가 및 모달 컴포넌트 수정

- 예약 상세 페이지에 취소 액션 섹션 추가
- 취소 모달 컴포넌트에서 건강상의 이유 옵션 주석 처리
- RejectionModal에 TODO 주석 추가하여 향후 매칭 로직 연동 필요성 명시

heroinesy committed:
refact: 매칭 결과 페이지 및 예약 상세 페이지 수정

- 예약 상세 페이지의 매니저 정보를 배열로 처리하도록 변경
- 매칭 결과 페이지에서 예약 취소 로직 수정
- 예약 상세 페이지에서 매칭 매니저 수 표시 추가

heroinesy committed:
Merge remote-tracking branch 'origin/develop' into feature/#60-manager-main

# Conflicts:
#	apps/src/app/(user)/(standalone)/reservation/[reservationId]/confirmation/page.tsx
#	apps/src/components/auth/CustomerAuthGuard.tsx

heroinesy committed:
feat: 수요자 예약 상세 페이지 데이터 페칭 방식 변경 및 스켈레톤 UI 추가

- check-in/out, comment 섹션을 예약 완료 상태시 데이터 표출
-ReservationDetailPageClient에서 서비스 완료 정보 섹션 추가 및 스타일 개선
- 불필요한 코드 및 주석 제거

heroinesy committed:
feat: 매니저 예약 상세 페이지 UI 개선 및 상태 배지 추가

- 예약 헤더 섹션에 상태 배지 컴포넌트 추가
- 예약일 및 예약 시간의 스타일 변경
- 서비스 상세 정보 섹션에서 고객 특이사항 표시 개선
- 결제 정보 섹션 추가 및 고객 정보 섹션 구현
- 불필요한 코드 정리 및 스타일 일관성 유지

Chung-Daeun reviewed (approved):
확인 완료!

재매칭 요청하는 부분은 매칭 수정 이후에 넣을지 말지 고민하신다는 말인거죠?

heroinesy committed:
feat: 추천 시간 API 호출 및 로딩 상태 관리 추가

- 예약 폼에서 추천 시간 및 주소 정보를 비동기로 가져오는 기능 구현
- 추천 시간 로딩 상태를 관리하여 사용자에게 피드백 제공
- 관련된 UI 컴포넌트에서 추천 시간 표시 방식 수정
- 추천 시간 데이터 구조를 개선하여 코드 일관성 유지

heroinesy commented:
> 확인 완료!
> 
> 재매칭 요청하는 부분은 매칭 수정 이후에 넣을지 말지 고민하신다는 말인거죠?

네네 마자요!!! 급하게 올린다고 저거 메모에 제가 막 적어둔 거 그대로 올려서ㅠㅠㅋㅋㅋㅋ,, 추후에 매칭 로직 구현한 다음에 환불로 할지 재매칭 로직 같이 붙일지 결정하려구용!

cheyounggi reviewed (approved):
잘 처리되었습니다.
